packages/core/workflows-sdk/src/utils/composer/create-step.ts

```diff
@@ -173,6 +173,10 @@ export function applyStep<
         ...localConfig,
       }
 
+      if (isDefined(newConfig.nested)) {
+        newConfig.nested ||= newConfig.async
```

Comment:
Is there any other config here that we'd like to store?

Reply:
I have also added some tests to validate the propagation of async and nested